packages/frontend/src/components/AudioRecorder/Encoder.ts

```diff
@@ -0,0 +1,112 @@
+/**
+ * This file is based on https://github.com/closeio/mic-recorder-to-mp3/blob/master/src/encoder.js
+ *
+ * @license MIT Copyright (c) 2017 Elastic Inc. (Close.io) see ./LICENSE
+ */
+
+import { Mp3Encoder } from '@breezystack/lamejs'
+
+interface EncoderConfig {
+  sampleRate?: number
+  bitRate?: number
+}
+
+class Encoder {
+  private config: Required<EncoderConfig>
+  private mp3Encoder: Mp3Encoder
+  private maxSamples: number
+  private samplesMono: Int16Array | null
+  private dataBuffer: Int8Array[] = []
+
+  constructor(config: EncoderConfig = {}) {
+    this.config = {
+      sampleRate: 44100,
+      bitRate: 128,
```

Comment:
this results in unexpectedly high data usage. until we can switch to opus, i suggest to half both parameters, for voice that is usually good enough

```suggestion
      sampleRate: 22050,
      bitRate: 64,
```

this should result in files being only 1/4 of size ([older considerations](https://github.com/deltachat/deltachat-android/issues/535#issuecomment-451418674))

moreover, we should respect the "media_quality" setting, but that can be done in another PR to not complicate things here further

Reply:
If we're into downsampling, for speech 8000 would be enough to make it understandable, but yeah, that is gonna be phone-level quality.

https://en.wikipedia.org/wiki/Sampling_(signal_processing)#Speech_sampling

For reference, in Telegram a one-minute voice message takes up ~230KB.

Reply:
I played around with various values but it didn't had any impact on the size of the recorded audio. It's always ~484kB for 30 seconds.

Reply:
hm, then there is a bug somewhere, maybe in the used library, ~1 mb/minute is the standard rate mp3 has as 128 kbit, 44.1 kHz 

i think, it should not block merging this PR, but we might want to file an issue for that

Reply:
I will open a new issue when this PR is merged